File: M3/UF5/e05_FrasesCelebres/src/Phrase.java
import java.util.ArrayList;
import java.util.List;

public class Phrase {
    private String text;
    private Author author;
    private List<Theme> themes;

    public Phrase(String text, Author author) {
        this.text = text;
        this.author = author;
        this.themes = new ArrayList<>();
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public Author getAuthor() {
        return author;
    }

    public List<Theme> getThemes() {
        return themes;
    }

    public void addTheme(Theme theme) {
        this.themes.add(theme);
    }
}

File: M7/myweb2/src/Frases/Phrase.php
<?php
namespace Frases\Entity;
use Doctrine\Common\Collections\ArrayCollection;

/**
 * @Entity
 * @Table(name="tbl_Phrases")
 */
class Phrase
{
    /**
     * @var int
     * @Id
     * @GeneratedValue
     * @Column(type="integer")
     */
    private $id;

    /**
     * @var string
     * @Column(type="string", name="text", nullable=false, unique=true)
     */
    private $text;

    /**
     * @ManyToOne(targetEntity="Author", inversedBy="phrases")
     * @JoinColumn(name="author_id", referencedColumnName="id")
     */
    private $author;

    /**
     * @ManyToMany(targetEntity="Theme", inversedBy="id", fetch="EAGER", cascade={"persist"}, orphanRemoval=true)
     * @JoinTable(
     *     name="phrase_theme",
     *     inverseJoinColumns={@JoinColumn(name="theme_id", referencedColumnName="id")}
     * )
     */
    private $themes;

    public function __construct()
    {
        $this->themes = new ArrayCollection();
    }

    public function addTheme(Theme $theme)
    {
        if (!$this->themes->contains($theme)) {
            $this->themes[] = $theme;
            $theme->addPhrase($this);
        }
        return $this;
    }

    // Métodos getter y setter
}